app/views/messages/_message.html.erb

```diff
@@ -32,14 +35,32 @@
       <% end %>
 
       <%# This div needs to be tightly wrapped around the content_text output because the whitespace formatting is important. %>
-      <div class="whitespace-pre-wrap" data-role="content-text"><%= message.content_text %></div>
+      <div class="whitespace-pre-wrap" data-role="content-text" data-clipboard-target="text"><%= message.content_text %></div>
 
       <div class="flex justify-start invisible gap-2 mt-2 text-gray-600 dark:text-gray-300 group-hover:visible">
         <% if message.user? %>
-          <%= icon "pencil", variant: :outline, size: 18, class: 'hover:text-gray-900 dark:hover:text-white cursor-pointer stroke-2', title: "Edit" %>
+          <%= button_tag type: "button",
+                class: "cursor-pointer hover:text-gray-900 dark:hover:text-white" do
+          %>
+            <%= icon "pencil", variant: :outline, size: 18, title: "Edit" %>
+          <% end %>
         <% else %>
-          <%= icon "clipboard", variant: :outline, size: 18, class: 'hover:text-gray-900 dark:hover:text-white cursor-pointer stroke-2', title: "Copy" %>
-          <%= icon "arrow-path", variant: :outline, size: 18, class: 'hover:text-gray-900 dark:hover:text-white cursor-pointer stroke-2', title: "Regenerate" %>
+          <%= button_tag type: "button",
+                class: "cursor-pointer hover:text-gray-900 dark:hover:text-white",
+                data: { action: %|
+                  clipboard#copy
+                  transition#toggleClassOn
+                  mouseleave->transition#toggleClassOff
+                | } do
+          %>
```

Comment:
I proceeded with converting these icons to buttons so that the data-action="" is a bit more natural, since we are triggering on click. While I was at it, I converted the other icons to buttons as well since we'll need those soon.